app/src/main/java/com/snaggly/ksw_toolkit/core/service/mcu/parser/LightEvent.kt

```diff
@@ -1,18 +1,31 @@
 package com.snaggly.ksw_toolkit.core.service.mcu.parser
 
+import android.content.Context
+import android.provider.Settings
 import com.snaggly.ksw_toolkit.core.service.mcu.McuLogic
 import com.snaggly.ksw_toolkit.core.service.mcu.parser.interfaces.ILightEvent
 import com.snaggly.ksw_toolkit.util.list.eventtype.EventManagerTypes
+import com.wits.pms.utils.BrightnessUtils
 import projekt.auto.mcu.ksw.serial.collection.McuCommands
 import kotlin.experimental.and
+import kotlin.math.roundToInt
+
 
 object LightEvent : ILightEvent {
-    override fun getCarDataEvent(data: ByteArray): EventManagerTypes {
+    override fun getCarDataEvent(data: ByteArray, context: Context): EventManagerTypes {
```

Comment:
Consider changing the LightEvent to a class type containing the context. That way you can create it once and you don't have to continue passing the context all around the code. See ScreenSwitchEventNoOEMScreen.kt for example: https://github.com/Snaggly/KSW-ToolKit-Service/blob/c0a0abd9490f5ad754f3d94499b51a9f6254cd31/app/src/main/java/com/snaggly/ksw_toolkit/core/service/mcu/parser/ScreenSwitchEventNoOEMScreen.kt#L13
And how it's created in McuReaderHandler.kt:
https://github.com/Snaggly/KSW-ToolKit-Service/blob/c0a0abd9490f5ad754f3d94499b51a9f6254cd31/app/src/main/java/com/snaggly/ksw_toolkit/core/service/mcu/McuReaderHandler.kt#L75

Reply:
ICarDataEvent() also used LightEvent(), so the context adaptation has increased, but it is no longer pulled around by getCarDataEvent(). Check after the push.